### PR TITLE
Add editorconfig with tuned configuration

### PR DESCRIPTION
**Pull request by Evangelink**
Add `.editorconfig` with a set of rules based on repository patterns and offline discussion.

### CODE REVIEW

File: src/coverlet.console/ConsoleTables/ConsoleTable.cs [hunk 1]
```diff
@@ -1,24 +1,5 @@
...
-// copies or substantial portions of the Software.
-// 
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-// SOFTWARE.
+﻿// Copyright (c) Toni Solarin-Sodara
```
Comment (Evangelink) [inline src/coverlet.console/ConsoleTables/ConsoleTable.cs:1]: Will revert this change.

File: src/coverlet.console/ConsoleTables/ConsoleTable.cs [hunk 2]
```diff
@@ -1,24 +1,5 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // ⚠ Do not modify this file. It will be replaced by a package reference once ConsoleTables has a strong name.
```
Comment (Evangelink) [inline src/coverlet.console/ConsoleTables/ConsoleTable.cs:5]: Do you want me to revert and exclude all changes to this file? Asking based on above comment ^^^.

**Comment by MarcoRossignoli** [comment]
We should keep tests project out of the refactoring. We tests coverage based on IL. Update the format will update the emitted IL.

If possible this project https://github.com/coverlet-coverage/coverlet/tree/master/test/coverlet.core.tests

**Comment by Evangelink** [comment]
@MarcoRossignoli PR is ready for review. Let me know if you want me to enable as many rules as possible here or merge this one and do follow-up PRs.

**Comment by MarcoRossignoli** [comment]
Thanks a lot I'll take a look asap

**Comment by daveMueller** [comment]
Didn't know that .NET analyzers and EditorConfig work hand-in-hand 🤔. I'm totally for this PR, definitely a good step to align `coverlet` code.

**Review by MarcoRossignoli — APPROVED** [review]
